### DISCUSSION TITLE
How do I setup the docker compose version?

### DISCUSSION THREAD

**Discussion by walter-bd** [testcontainers/testcontainers-node#615] [Q&A] [answered]
Hi,

I want to know where can I setup the docker compose version when using DockerComposeEnvironment, because my current development environment is configured for docker compose v2 but it appears that testcontainers defaults to docker compose v1.29. Which leads to this error:
`Error: spawn docker-compose ENOENT`

If someone knows how to configure this, it would be of great help.

My environment is:
docker compose: 2.19.1
node: 18.16.1 
testcontainers: 9.11
running on Ubuntu  22.04

Thank you.

**Comment by cristianrgreco** [comment]
Hi @walter-bd, would you mind sharing the logs of a test run? You can enable logs by exporting the `DEBUG=testcontainers*` environment variable. I'd expect Testcontainers to default to v1.29 only if it detects that it exists. Locally I also have docker-compose v2.19.1 and the tests pass as expected.

**Reply by walter-bd** [reply]
Sure, here it is

  testcontainers [INFO] Starting DockerCompose environment "testcontainers-fe4a5fa51fb0"... +0ms
  testcontainers [DEBUG] Loading ".testcontainers.properties" file... +1ms
  testcontainers [DEBUG] Testing Docker client strategy "UnixSocketStrategy" with URI "unix:///var/run/docker.sock"... +2ms
  testcontainers [DEBUG] Fetching system info... +9ms
  testcontainers [INFO] Unable to detect docker-compose version, is it installed? Error: spawn docker-compose ENOENT +18ms
  testcontainers [DEBUG] Node version: v18.16.1, Platform: linux, Arch: x64, OS: Ubuntu 22.04.2 LTS, Version: 24.0.4, Arch: x86_64, CPUs: 12, Memory: 16456708096, Compose installed: false, Compose version: N/A +0ms
  testcontainers [DEBUG] Acquiring lock file "/tmp/tc.lock"... +2ms
  testcontainers [DEBUG] Acquired lock file "/tmp/tc.lock" +3ms
  testcontainers [DEBUG] Creating new Reaper for session "3086a77735a6" with socket path "/var/run/docker.sock"... +3ms
  testcontainers [DEBUG] Not pulling image "testcontainers/ryuk:0.5.1" as it already exists +4ms
  testcontainers [INFO] Creating container for image "testcontainers/ryuk:0.5.1"... +1ms
  testcontainers [INFO] [7d1550944c11] Created container for image "testcontainers/ryuk:0.5.1" +14ms
  testcontainers [INFO] [7d1550944c11] Starting container for image "testcontainers/ryuk:0.5.1"... +0ms
  testcontainers [INFO] [7d1550944c11] Started container for image "testcontainers/ryuk:0.5.1" +274ms
  testcontainers [DEBUG] [7d1550944c11] Waiting for container to be ready... +4ms
  testcontainers [DEBUG] [7d1550944c11] Waiting for log message "/.+ Started!/"... +0ms
  testcontainers:containers [7d1550944c11] 2023/07/19 14:01:30 Pinging Docker... +0ms
  testcontainers:containers [7d1550944c11] 2023/07/19 14:01:30 Docker daemon is available! +0ms
  testcontainers:containers [7d1550944c11] 2023/07/19 14:01:30 Starting on port 8080... +0ms
  testcontainers:containers [7d1550944c11] 2023/07/19 14:01:30 Started! +0ms
  testcontainers [INFO] [7d1550944c11] Container is ready +5ms
  testcontainers [DEBUG] [7d1550944c11] Connecting to Reaper (attempt 1) on "localhost:32780"... +1ms
  testcontainers [DEBUG] [7d1550944c11] Connected to Reaper +1ms
  testcontainers [DEBUG] Releasing lock file "/tmp/tc.lock"... +1ms
  testcontainers:containers [7d1550944c11] 2023/07/19 14:01:30 New client connected: 172.17.0.1:44490 +7ms
  testcontainers:containers [7d1550944c11] 2023/07/19 14:01:30 Adding {"label":{"org.testcontainers.session-id=3086a77735a6":true}} +0ms
  testcontainers [DEBUG] Released lock file "/tmp/tc.lock" +0ms
  testcontainers [INFO] Docker client strategy "UnixSocketStrategy" works, localhost (127.0.0.1) +1ms
  testcontainers [INFO] Upping DockerCompose environment... +0ms
  testcontainers [ERROR] Failed to up DockerCompose environment: spawn docker-compose ENOENT +6ms
  testcontainers [INFO] Downing DockerCompose environment... +1ms
  testcontainers [ERROR] Failed to down DockerCompose environment: spawn docker-compose ENOENT +5ms
Error: spawn docker-compose ENOENT

As you see is failing to detect docker compose v2, which I have installed in my environment

**Reply by cristianrgreco** [reply]
> testcontainers [INFO] Unable to detect docker-compose version, is it installed? Error: spawn docker-compose ENOENT +18ms

This is what I was hoping for. It is because v1 is the default. I may be able to automatically fallback to trying v2 when this happens. Will keep you posted, thanks!

**Reply by walter-bd** [reply]
Thank you!

**Comment by cristianrgreco** [comment, marked as answer]
Docker compose version is automatically detected since version 9.12.0. 
Testcontainers will use the `docker-compose` or `docker compose` binaries.

**Reply by walter-bd** [reply]
Nice, thanks.
